### DISCUSSION TITLE
Impossible to create styles for each component

### DISCUSSION THREAD

**Discussion by mlbonniec** [vercel/next.js#13109] [Help]
# Bug report
All styles must be included in `_app.js` and therefore be global. I include my `.scss` files, but they must be global. 
It seems that this tutorial is what I'm looking for, but you have to add a variable as a class to make it work, and as indicated in the post, it's no longer compatible with the new versions of Next.JS https://medium.com/@vladymyr.pylypchatin/the-simple-way-to-use-scoped-and-global-scss-with-next-js-67cdb2d0c676
I also tried `@webdeb/next-styles` but the styles are not unique to the component. They are applied by all HTML elements, even when the component is not displayed.

## Expected behavior
I would like to be able to use `.scss` files to be included in every component and page, so that I can use the style only when the component is rendered in the page.

## System information
- OS: macOS
- Browser: Firefox
- Version of Next.js: Latest, 9.4.1
- Version of Node.js: v12.16.1

I need help, thanks

**Comment by jamesmosier** [comment]
There are numerous issues like this already:

https://github.com/zeit/next.js/issues/12925
https://github.com/zeit/next.js/issues/12079

#12079 will cover what you are mentioning.

**Comment by mlbonniec** [comment]
Ok I see, but with `next/sass` or `next-styles` I was able to import my styles into the `.js` file of the component.
However, the style also applies to other elements, even when the component is not present.
How can I make the component style apply only when the component is displayed?
In the manner of a `<style jsx></style>` tag that inserts itself into the `<head>`

**Comment by jamesmosier** [comment]
So you are looking for component scoped scss?

```jsx
import app from '../app.module.scss'
function Index() {
  return (
    <div className={app.foo}>hi</div>
  )
}
```

`app.module.scss`

```css
.foo {
  background: red;
}
```

**Reply by ankit-gupta1307** [reply]
Hi @jamesmosier I want to use it in a similar fashion but my nested scss is not working. If you are aware then please reply.

**Reply by ankit-gupta1307** [reply]
Also, when I am using nested scss globally(i.e by importing the scss file in _app.js) then it's working.

**Reply by jamesmosier** [reply]
hi @ankit-gupta1307! feel free to open up a new Discussion with a code sample of what you are looking to do! that might be easiest to get all the information about what you are trying to accomplish 😄

**Reply by ankit-gupta1307** [reply]
Okay, thanks for the quick reply.

**Reply by ankit-gupta1307** [reply]
@jamesmosier Please check this #14333

**Comment by mlbonniec** [comment]
Yeah, that's pretty much what I want. But isn't there a way to do this without adding the variable as a class?
Let's say I want to put a `background-color: red;` to the `<body></body>`, just on a specific page. My body tag is directly in `_document.js`. So I can't use scoped styles ?

**Comment by smurrayatwork** [comment]
For anyone else that's trying to use this with dart sass' js implementation for things like `@use` support and sass modules, if you have -any- other node module that has a dependency on node-sass, the default next setup will use node-sass instead of sass. Locally I've fixed it by doing the following:
```
// example next.config.js
module.exports = {
webpack(config, options) {
  config.module.rules.forEach(rule => {
          if (rule.oneOf) {
            const nestedScss = rule.oneOf.find((one) => {
              return one.test
                && 'some.scss'.match(one.test) 
                && one.issuer 
                && one.issuer.include 
                && one.issuer.include.includes('_app');
            });
            if (nestedScss) {
              const sassLoader = nestedScss.use.find(u => u.loader.includes('sass-loader'));
              // Set implementation to sass instead of node-sass here.
              sassLoader.options.implementation = require('sass');
            }
          }
        })
  }
}
```

You'll then need to import your scss files in `_app.js`.
